CDP4Common/AutoGenEquatable/ActionItemEquatable.cs

```diff
@@ -0,0 +1,159 @@
+ | 15    | languageCode                         | string                       | 1..1        |  1.1.0  |
+ | 16    | modifiedOn                           | DateTime                     | 1..1        |  1.1.0  |
+ | 17    | owner                                | Guid                         | 1..1        |  1.1.0  |
+ | 18    | primaryAnnotatedThing                | Guid                         | 0..1        |  1.1.0  |
+ | 19    | relatedThing                         | Guid                         | 0..*        |  1.1.0  |
+ | 20    | shortName                            | string                       | 1..1        |  1.1.0  |
+ | 21    | sourceAnnotation                     | Guid                         | 0..*        |  1.1.0  |
+ | 22    | status                               | AnnotationStatusKind         | 1..1        |  1.1.0  |
+ | 23    | title                                | string                       | 1..1        |  1.1.0  |
+ | 24    | thingPreference                      | string                       | 0..1        |  1.2.0  |
+ * -------------------------------------------- | ---------------------------- | ----------- | ------- */
+
+namespace CDP4Common.DTO.Equatable
+{
+    using System.Linq;
+
+    using CDP4Common.DTO;
+
+    /// <summary>
+    /// An extension class for the <see cref="ActionItem"/> to support the computation
+    /// of property based equality
+    /// </summary>
+    public static class ActionItemEquatable
+    {
+        /// <summary>
+        /// Determines whether 2 <see cref="ActionItem"/> objects are the same
+        /// by comparing all the properties
+        /// </summary>
+        /// <param name="me">
+        /// The current <see cref="ActionItem"/>
+        /// </param>
+        /// <param name="other">
+        /// The <see cref="ActionItem"/> that the <paramref name="me"/> is
+        /// compared to
+        /// </param>
+        /// <returns>
+        /// returns true when all the properties of the <see cref="ActionItem"/> objects
+        /// have the same value
+        /// </returns>
+        public static bool ArePropertiesEqual(this ActionItem me, ActionItem other)
+        {
+            if (me == null && other == null) return true;
+
+            if (me == null || other == null) return false;
+
+            if (!me.Iid.Equals(other.Iid)) return false;
+
+            if (!me.RevisionNumber.Equals(other.RevisionNumber)) return false;
+
+            if (!me.Actionee.Equals(other.Actionee)) return false;
+
+            if (!me.ApprovedBy.OrderBy(x => x).SequenceEqual(other.ApprovedBy.OrderBy(x => x))) return false;
+
+            if (!me.Author.Equals(other.Author)) return false;
+
+            if (!me.Category.OrderBy(x => x).SequenceEqual(other.Category.OrderBy(x => x))) return false;
+
+            if (!me.Classification.Equals(other.Classification)) return false;
+
+            if (!me.CloseOutDate.HasValue && other.CloseOutDate.HasValue) return false;
```

Comment:
does it maybe not make more sense to have something like:
`if (me.CloseOutDate.HasValue != other.CloseOutDate.HasValue) return false;`

Because right now the logic is "If I dont have value and you have value" but doesnt cover "If I have value and you don't have value"